--- nemo_orig/debian/control	2020-02-02 23:26:51.768329800 +0200
+++ nemo/debian/control	2020-02-02 23:14:03.337983179 +0200
@@ -3,13 +3,12 @@
 Priority: optional
 Maintainer: Linux Mint <root@linuxmint.com>
 Build-Depends:
- cinnamon-l10n,
  debhelper (>= 10),
  gobject-introspection,
  gtk-doc-tools (>= 1.4),
  intltool (>= 0.40.1),
  libatk1.0-dev (>= 1.32.0),
- libcinnamon-desktop-dev (>= 3.6),
+ libmate-desktop-dev,
  libexempi-dev (>= 2.2.0),
  libexif-dev (>= 0.6.20),
  libgail-3-dev,
@@ -81,12 +80,9 @@
 Architecture: any
 Pre-Depends: ${misc:Pre-Depends}
 Depends:
- cinnamon-desktop-data,
- cinnamon-l10n,
  desktop-file-utils (>= 0.7),
  gsettings-desktop-schemas,
  gvfs (>= 1.3.2),
- libcinnamon-desktop4 (>= 2.6.1),
  libglib2.0-data,
  libnemo-extension1 (= ${binary:Version}),
  libxapp1 (>= 1.4.0),
diff -Naur '--exclude=.git' nemo_orig/libnemo-private/meson.build nemo/libnemo-private/meson.build
--- nemo_orig/libnemo-private/meson.build	2020-02-02 22:20:50.619461010 +0200
+++ nemo/libnemo-private/meson.build	2020-02-02 12:47:31.414009875 +0200
@@ -84,7 +84,7 @@
 ]
 
 nemo_private_deps = [
-  cinnamon, eel, gail, glib, gmodule, gtk, libxml, math, nemo_extension, x11,
+  mate, eel, gail, glib, gmodule, gtk, libxml, math, nemo_extension, x11,
 ]
 
 if libexif_enabled
diff -Naur '--exclude=.git' nemo_orig/libnemo-private/nemo-thumbnails.c nemo/libnemo-private/nemo-thumbnails.c
--- nemo_orig/libnemo-private/nemo-thumbnails.c	2020-02-02 22:20:50.619461010 +0200
+++ nemo/libnemo-private/nemo-thumbnails.c	2020-02-02 15:27:50.958524186 +0200
@@ -26,7 +26,7 @@
 #include <config.h>
 #include "nemo-thumbnails.h"
 
-#define GNOME_DESKTOP_USE_UNSTABLE_API
+// #define GNOME_DESKTOP_USE_UNSTABLE_API
 
 #include "nemo-directory-notify.h"
 #include "nemo-global-preferences.h"
@@ -43,7 +43,9 @@
 #include <sys/wait.h>
 #include <unistd.h>
 #include <signal.h>
-#include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+// #include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+#define MATE_DESKTOP_USE_UNSTABLE_API
+#include <libmate-desktop/mate-desktop-thumbnail.h>
 
 #define DEBUG_FLAG NEMO_DEBUG_THUMBNAILS
 #include <libnemo-private/nemo-debug.h>
@@ -61,9 +63,9 @@
 #define NEMO_THUMBNAIL_FRAME_RIGHT 3
 #define NEMO_THUMBNAIL_FRAME_BOTTOM 3
 
-/* structure used for making thumbnails, associating a uri with where the thumbnail is to be stored */
+    /* structure used for making thumbnails, associating a uri with where the thumbnail is to be stored */
 
-typedef struct {
+    typedef struct {
     char *image_uri;
     char *mime_type;
     time_t original_file_mtime;
@@ -99,7 +101,7 @@
  * to avoid adding it again. Lock thumbnails_mutex when accessing this. */
 static NemoThumbnailInfo *currently_thumbnailing = NULL;
 
-static GnomeDesktopThumbnailFactory *thumbnail_factory = NULL;
+static MateDesktopThumbnailFactory *thumbnail_factory = NULL;
 
 static gboolean
 get_file_mtime (const char *file_uri, time_t* mtime)
@@ -134,11 +136,11 @@
     g_free (info);
 }
 
-static GnomeDesktopThumbnailFactory *
+static MateDesktopThumbnailFactory *
 get_thumbnail_factory (void)
 {
     if (thumbnail_factory == NULL) {
-        thumbnail_factory = gnome_desktop_thumbnail_factory_new (GNOME_DESKTOP_THUMBNAIL_SIZE_LARGE);
+        thumbnail_factory = mate_desktop_thumbnail_factory_new (MATE_DESKTOP_THUMBNAIL_SIZE_LARGE);
     }
 
     return thumbnail_factory;
@@ -304,7 +306,7 @@
 gboolean
 nemo_can_thumbnail (NemoFile *file)
 {
-    GnomeDesktopThumbnailFactory *factory;
+    MateDesktopThumbnailFactory *factory;
     gboolean res;
     char *uri;
     time_t mtime;
@@ -315,7 +317,7 @@
     mtime = nemo_file_get_mtime (file);
     
     factory = get_thumbnail_factory ();
-    res = gnome_desktop_thumbnail_factory_can_thumbnail (factory,
+    res = mate_desktop_thumbnail_factory_can_thumbnail (factory,
                                  uri,
                                  mime_type,
                                  mtime);
@@ -552,18 +554,18 @@
                        info->image_uri);
         }
 
-        pixbuf = gnome_desktop_thumbnail_factory_generate_thumbnail (thumbnail_factory,
+        pixbuf = mate_desktop_thumbnail_factory_generate_thumbnail (thumbnail_factory,
                                          info->image_uri,
                                          info->mime_type);
 
         if (pixbuf) {
-            gnome_desktop_thumbnail_factory_save_thumbnail (thumbnail_factory,
+            mate_desktop_thumbnail_factory_save_thumbnail (thumbnail_factory,
                                     pixbuf,
                                     info->image_uri,
                                     current_orig_mtime);
             g_object_unref (pixbuf);
         } else {
-            gnome_desktop_thumbnail_factory_create_failed_thumbnail (thumbnail_factory, 
+            mate_desktop_thumbnail_factory_create_failed_thumbnail (thumbnail_factory, 
                                          info->image_uri,
                                          current_orig_mtime);
         }
@@ -707,8 +709,8 @@
     g_mutex_unlock (&thumbnails_mutex);
 }
 
-gboolean
-nemo_thumbnail_factory_check_status (void)
-{
-    return gnome_desktop_thumbnail_cache_check_permissions (get_thumbnail_factory (), TRUE);
-}
+// gboolean
+// nemo_thumbnail_factory_check_status (void)
+// {
+//     return mate_desktop_thumbnail_cache_check_permissions (get_thumbnail_factory (), TRUE);
+// }
diff -Naur '--exclude=.git' nemo_orig/meson.build nemo/meson.build
--- nemo_orig/meson.build	2020-02-02 22:20:50.619461010 +0200
+++ nemo/meson.build	2020-02-02 12:47:24.061559880 +0200
@@ -78,7 +78,8 @@
 go_intr = dependency('gobject-introspection-1.0',   version: '>=1.0')
 libnotif= dependency('libnotify',                   version: '>=0.7.0')
 
-cinnamon= dependency('cinnamon-desktop',            version: '>=2.6.1')
+# cinnamon= dependency('cinnamon-desktop',            version: '>=2.6.1')
+mate    = dependency('mate-desktop-2.0')
 gail    = dependency('gail-3.0')
 libxml  = dependency('libxml-2.0',                  version: '>=2.7.8')
 x11     = dependency('x11')
diff -Naur '--exclude=.git' nemo_orig/src/meson.build nemo/src/meson.build
--- nemo_orig/src/meson.build	2020-02-02 22:20:50.619461010 +0200
+++ nemo/src/meson.build	2020-02-02 12:47:28.957861185 +0200
@@ -102,7 +102,7 @@
   nemoCommon_sources += 'nemo-empty-view.c'
 endif
 
-nemo_deps = [ cinnamon, gail, glib, gtk, libnotif, libxml, math, x11,
+nemo_deps = [ mate, gail, glib, gtk, libnotif, libxml, math, x11,
   egg, nemo_extension, nemo_private, xapp ]
 
 if exempi_enabled
diff -Naur '--exclude=.git' nemo_orig/src/nemo-application.c nemo/src/nemo-application.c
--- nemo_orig/src/nemo-application.c	2020-02-02 22:20:50.619461010 +0200
+++ nemo/src/nemo-application.c	2020-02-02 15:23:54.282267803 +0200
@@ -77,8 +77,10 @@
 #include <eel/eel-string.h>
 #include <libnotify/notify.h>
 
-#define GNOME_DESKTOP_USE_UNSTABLE_API
-#include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+// #define GNOME_DESKTOP_USE_UNSTABLE_API
+// #include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+#define MATE_DESKTOP_USE_UNSTABLE_API
+#include <libmate-desktop/mate-desktop-thumbnail.h>
 
 #define NEMO_ACCEL_MAP_SAVE_DELAY 30
 
@@ -560,13 +562,13 @@
     /* silently do a full check of the cache and fix if running as root.
      * If running as a normal user, do a quick check, and we'll notify the
      * user later if there's a problem via an infobar */
-    if (geteuid () == 0) {
-        if (!gnome_desktop_thumbnail_cache_check_permissions (NULL, FALSE))
-            gnome_desktop_thumbnail_cache_fix_permissions ();
-    } else {
-        if (!gnome_desktop_thumbnail_cache_check_permissions (NULL, TRUE))
-            self->priv->cache_problem = TRUE;
-    }
+    // if (geteuid () == 0) {
+    //     if (!mate_desktop_thumbnail_cache_check_permissions (NULL, FALSE))
+    //         mate_desktop_thumbnail_cache_fix_permissions ();
+    // } else {
+    //     if (!mate_desktop_thumbnail_cache_check_permissions (NULL, TRUE))
+    //         self->priv->cache_problem = TRUE;
+    // }
 }
 
 static void
@@ -639,11 +641,11 @@
     return singleton;
 }
 
-void
-nemo_application_check_thumbnail_cache (NemoApplication *application)
-{
-    application->priv->cache_problem = !nemo_thumbnail_factory_check_status ();
-}
+// void
+// nemo_application_check_thumbnail_cache (NemoApplication *application)
+// {
+//     application->priv->cache_problem = !nemo_thumbnail_factory_check_status ();
+// }
 
 gboolean
 nemo_application_get_cache_bad (NemoApplication *application)
diff -Naur '--exclude=.git' nemo_orig/src/nemo-bookmark-list.c nemo/src/nemo-bookmark-list.c
--- nemo_orig/src/nemo-bookmark-list.c	2020-02-02 22:20:50.623460839 +0200
+++ nemo/src/nemo-bookmark-list.c	2020-02-02 14:17:31.841691491 +0200
@@ -31,8 +31,51 @@
 #include <libnemo-private/nemo-file-utilities.h>
 #include <libnemo-private/nemo-file.h>
 #include <libnemo-private/nemo-icon-names.h>
-#define GNOME_DESKTOP_USE_UNSTABLE_API
-#include <libcinnamon-desktop/gnome-desktop-utils.h>
+// #define GNOME_DESKTOP_USE_UNSTABLE_API
+#define MATE_DESKTOP_USE_UNSTABLE_API
+#include <libmate-desktop/mate-desktop-utils.h>
+
+// This all came from gnome-desktop-utils.h from libcinnamon-desktop
+#include <pwd.h>    
+
+struct passwd *
+mate_desktop_get_session_user_pwent_(void)
+{
+    struct passwd *pwent = NULL;
+
+    if (getuid() != geteuid())
+    {
+        gint uid = getuid();
+        pwent = getpwuid(uid);
+    }
+    else if (g_getenv("SUDO_UID") != NULL)
+    {
+        gint uid = (int)g_ascii_strtoll(g_getenv("SUDO_UID"), NULL, 10);
+        pwent = getpwuid(uid);
+    }
+    else if (g_getenv("PKEXEC_UID") != NULL)
+    {
+        gint uid = (int)g_ascii_strtoll(g_getenv("PKEXEC_UID"), NULL, 10);
+        pwent = getpwuid(uid);
+    }
+    else if (g_getenv("USERNAME") != NULL)
+    {
+        pwent = getpwnam(g_getenv("USERNAME"));
+    }
+    else if (g_getenv("USER") != NULL)
+    {
+        pwent = getpwnam(g_getenv("USER"));
+    }
+
+    if (!pwent)
+    {
+        return getpwuid(getuid());
+    }
+
+    return pwent;
+}
+
+/* end of hacks */
 
 #include <gio/gio.h>
 #include <string.h>
@@ -66,7 +109,7 @@
 {
     if (geteuid () == 0) {
         struct passwd *pwent;
-        pwent = gnome_desktop_get_session_user_pwent ();
+        pwent = mate_desktop_get_session_user_pwent_ ();
 
         gchar *path = g_file_get_path (file);
 
diff -Naur '--exclude=.git' nemo_orig/src/nemo-main-application.c nemo/src/nemo-main-application.c
--- nemo_orig/src/nemo-main-application.c	2020-02-02 22:20:50.623460839 +0200
+++ nemo/src/nemo-main-application.c	2020-02-02 14:15:30.677484874 +0200
@@ -72,9 +72,10 @@
 #include <eel/eel-stock-dialogs.h>
 #include <libnotify/notify.h>
 
-#define GNOME_DESKTOP_USE_UNSTABLE_API
-
-#include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+// #define GNOME_DESKTOP_USE_UNSTABLE_API
+// #include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+#define MATE_DESKTOP_USE_UNSTABLE_API
+#include <libmate-desktop/mate-desktop-thumbnail.h>
 
 /* Keep window from shrinking down ridiculously small; numbers are somewhat arbitrary */
 #define APPLICATION_WINDOW_MIN_WIDTH	300
@@ -654,7 +655,7 @@
         if (geteuid () != 0) {
             g_printerr ("The --fix-cache option must be run with sudo or as the root user.\n");
         } else {
-            gnome_desktop_thumbnail_cache_fix_permissions ();
+            // gnome_desktop_thumbnail_cache_fix_permissions ();
             g_print ("User thumbnail cache successfully repaired.\n");
         }
 
diff -Naur '--exclude=.git' nemo_orig/src/nemo-properties-window.c nemo/src/nemo-properties-window.c
--- nemo_orig/src/nemo-properties-window.c	2020-02-02 22:20:50.623460839 +0200
+++ nemo/src/nemo-properties-window.c	2020-02-02 13:19:17.407298384 +0200
@@ -38,8 +38,10 @@
 #include <cairo.h>
 #include <libxapp/xapp-icon-chooser-dialog.h>
 
-#define GNOME_DESKTOP_USE_UNSTABLE_API
-#include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+// #define GNOME_DESKTOP_USE_UNSTABLE_API
+// #include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+#define MATE_DESKTOP_USE_UNSTABLE_API
+#include <libmate-desktop/mate-desktop-thumbnail.h>
 
 #include <eel/eel-accessibility.h>
 #include <eel/eel-glib-extensions.h>
diff -Naur '--exclude=.git' nemo_orig/src/nemo-thumbnail-problem-bar.c nemo/src/nemo-thumbnail-problem-bar.c
--- nemo_orig/src/nemo-thumbnail-problem-bar.c	2020-02-02 22:20:50.623460839 +0200
+++ nemo/src/nemo-thumbnail-problem-bar.c	2020-02-02 14:18:49.869753587 +0200
@@ -60,7 +60,7 @@
     switch (response_id) {
         case FIX_CACHE:
             g_spawn_command_line_sync ("sh -c \"pkexec nemo --fix-cache\"", NULL, NULL, NULL, NULL);
-            nemo_application_check_thumbnail_cache (nemo_application_get_singleton ());
+            // nemo_application_check_thumbnail_cache (nemo_application_get_singleton ());
             nemo_window_slot_queue_reload (nemo_view_get_nemo_window_slot (bar->priv->view), FALSE);
             nemo_window_slot_check_bad_cache_bar (nemo_view_get_nemo_window_slot (bar->priv->view));
             break;
